application/models/TemplateConfig.php

```diff
@@ -885,6 +885,74 @@ public static function uninstall($templatename)
         return false;
     }
 
+    /**
+     * Uninstalls all surveythemes that are being extended from the supplied surveytheme name
+     * @param $templateName
+     * @return void
+     * @throws CDbException
+     */
+    public static function uninstallThemesRecursive($templateName): void
```

Comment:
Should be in service class, or...? Only way we can add unit-tests in the future.

Reply:
I would like to but this hole setup with 3 different classes is already to much if i now also at a serviceclass everything might get more confusing it should all be one class with service/helper classes.